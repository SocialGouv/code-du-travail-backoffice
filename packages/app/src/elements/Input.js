import styled from "@emotion/styled";
import { omit } from "ramda";
import React from "react";

import Icon from "./Icon";

const InputWithIconWrapper = styled.div`
  margin-top: -1.375rem;
`;

const _Input = styled.input`
  -webkit-appearance: none;
  background-color: white;
  border: solid 1px var(${p => (p.hasError ? "--color-text-red" : "--color-border")});
  color: inherit;
  flex-grow: 1;
  font-family: inherit;
  font-weight: inherit;
  font-size: 0.875rem;
  height: 2rem;
  line-height: 1;
  opacity: ${p => (p.disabled ? 0.25 : 1)};
  padding: 0.5rem 0.6rem 0.55rem ${p => (p.hasIcon ? "2rem" : "0.6rem")};
  width: 100%;

  ::placeholder {
    color: var(--color-placeholder);
  }
`;
const InputIcon = styled(Icon)`
  color: var(--color-placeholder);
  left: 0.625rem;
  top: 1.775rem;
  position: relative;
`;

const Input = ({ disabled = false, icon, hasError, ...props }, ref) => {
  if (icon === undefined) {
    return <_Input disabled={disabled} hasError={hasError} hasIcon={false} ref={ref} {...props} />;
  }

  const { className, style } = props;
  const inputProps = omit(["className", "style"], props);

  return (
    <div className={className} style={style}>
      <InputWithIconWrapper>
        <InputIcon icon={icon} />
        <_Input disabled={disabled} hasError={hasError} hasIcon ref={ref} {...inputProps} />
      </InputWithIconWrapper>
    </div>
  );
};

// https://reactjs.org/docs/forwarding-refs.html
export default React.forwardRef(Input);
