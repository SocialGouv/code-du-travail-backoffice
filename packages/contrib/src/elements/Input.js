import React from "react";
import styled from "styled-components";

import Icon from "./Icon";

const Input = styled.input`
  -webkit-appearance: none;
  background-color: transparent;
  border: solid 1px var(--color-border);
  color: inherit;
  flex-grow: 1;
  font-family: inherit;
  font-weight: inherit;
  font-size: 0.875rem;
  height: 2rem;
  line-height: 1;
  padding: 0.4rem 0.6rem 0.575rem
    ${({ hasIcon }) => (Boolean(hasIcon) ? "2rem" : "0.6rem")};
  width: 100%;

  ::placeholder {
    color: var(--color-placeholder);
  }
`;
const InputIconContainer = styled.span`
  margin-left: 0.625rem;
  margin-top: 0.75rem;
  position: absolute;
`;
const InputIcon = styled(Icon)`
  color: var(--color-placeholder);
`;

export default ({ icon, ...props }) => {
  if (icon === undefined) return <Input {...props} />;

  return (
    <div>
      <InputIconContainer>
        <InputIcon icon={icon} />
      </InputIconContainer>
      <Input hasIcon {...props} />
    </div>
  );
};
