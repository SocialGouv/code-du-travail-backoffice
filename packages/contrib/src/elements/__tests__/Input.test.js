import React from "react";
import { render } from "react-testing-library";
import Input from "../Input";

describe("[Contrib] elements/<Input />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });
});
