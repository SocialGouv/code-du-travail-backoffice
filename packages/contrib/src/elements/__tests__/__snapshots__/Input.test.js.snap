// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<Input /> should match snapshot 1`] = `
<div>
  <input
    class="Input___Input-l1y0rz-1 bLtEdt"
  />
</div>
`;

exports[`[Contrib] elements/<Input /> should match snapshot with \`disabled\` prop 1`] = `
<div>
  <input
    class="Input___Input-l1y0rz-1 jHgeEG"
    disabled=""
  />
</div>
`;

exports[`[Contrib] elements/<Input /> should match snapshot with a custom \`icon\` prop 1`] = `
<div>
  <div>
    <div
      class="Input__InputWithIconWrapper-l1y0rz-0 Qjren"
    >
      <span
        class="Input__InputIconContainer-l1y0rz-2 IyOKF"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-search fa-w-16 fa-1x Input__InputIcon-l1y0rz-3 eiJHjO Icon-sc-4jlog9-0 hXLKsx"
          color="var(--color-black-leather-jacket)"
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
          />
        </svg>
      </span>
      <input
        class="Input___Input-l1y0rz-1 eMUdic"
      />
    </div>
  </div>
</div>
`;
